api/envoy/api/v2/core/base.proto

```diff
@@ -76,10 +77,43 @@ message Locality {
   string sub_zone = 3;
 }
 
+// BuildVersion combines SemVer version of extension with free-form build information
+// (i.e. 'alpha', 'private-build') as a set of strings.
+message BuildVersion {
+  // SemVer version of extension
+  type.SemanticVersion version = 1;
+
+  // Free-form build information i.e. 'alpha'
+  repeated string labels = 2;
+}
+
+// Version and identification for an Envoy extension.
+message Extension {
+  // This is the name of the Envoy filter as specified in the Envoy
+  // configuration, e.g. envoy.router, com.acme.widget.
+  string name = 1;
+
+  // Category of the extension
```

Comment:
Can you add docs on where to find these names and some examples?

Reply:
Good point. Right now the category comes from the factory's category() method. There is no single place where all categories are listed. Also not categories right now do not follow the DNS naming. I.e. tracers use the "tracing" category.
Some things to consider:
1.  Should we consolidate all category names into a common header?
1.  Should we add a notion of a deprecated category name and rename all non-conforming categories to use DNS notation?

Reply:
I think we could move towards (2), probably file an issue. For now, I think if we could do (1) with decent comments that would be awesome, we could just provide the GitHub URL to that header.

Reply:
PR #9425